packages/suite/src/components/firmware/FirmwareProgressBar.tsx

```diff
@@ -2,42 +2,33 @@ import styled, { useTheme } from 'styled-components';
 
 import { FirmwareOperationStatus } from '@suite-common/firmware';
 import { TranslationKey } from '@suite-common/intl-types';
-import { Box, Column, Icon, ProgressBar, Row, Text } from '@trezor/components';
+import { Box, Column, ProgressBar, Row, Text } from '@trezor/components';
 import { spacings } from '@trezor/theme';
 
 import { Translation } from 'src/components/suite/Translation';
 import { useFirmwareDesktopUpdate } from 'src/hooks/suite/useFirmwareDesktopUpdate';
 
-import { useSelector } from '../../hooks/suite';
-
 const Percentage = styled.div`
     font-variant-numeric: tabular-nums;
     width: 30px;
 `;
 
 export const FirmwareProgressBar = () => {
     const theme = useTheme();
-    const { operation, progress, uiEvent } = useFirmwareDesktopUpdate();
-    const isActiveOnboarding = useSelector(state => state.onboarding.isActive);
+    const { cachedDevice, operation, progress } = useFirmwareDesktopUpdate();
 
     const mapOperationToTranslationId: Record<
         NonNullable<FirmwareOperationStatus['operation']>,
         TranslationKey
     > = {
         installing: 'TR_INSTALLING',
-        restarting: isActiveOnboarding
+        restarting: !cachedDevice?.features?.pin_protection
             ? 'TR_RESTARTING_TREZOR'
-            : 'TR_RESTARTING_TREZOR_ENTER_PIN_IF_NEEDED',
+            : 'TR_RESTARTING_TREZOR_ENTER_PIN_WHEN_NEEDED',
```

Comment:
This string is no longer needed because it was replaced by the confirmation modal.

Reply:
will leave just `TR_RESTARTING_TREZOR`